lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -3423,6 +3423,20 @@ Value FIRRTLLowering::createArrayIndexing(Value array, Value index) {
   auto size = hw::type_cast<hw::ArrayType>(array.getType()).getSize();
   auto valWire = builder.create<sv::WireOp>(
       hw::type_cast<hw::ArrayType>(array.getType()).getElementType());
+
+  // Extend to power of 2
```

Comment:
Can the PR expand this comment to indicate why we're doing this? The PR may want to include the verbose `TODO` as well. It's not clear that we want this to be the final say here.